dashboard/test/controllers/home_controller_test.rb

```diff
@@ -11,14 +11,24 @@ class HomeControllerTest < ActionController::TestCase
     Properties.stubs(:get).returns nil
   end
 
-  test "redirect index when signed in" do
+  test "redirect index when signed in (if not student with course progress)" do
     user = create(:user)
     sign_in user
     get :index
 
     assert_redirected_to '/home'
   end
 
+  test "redirect next lesson if student with course progress" do
```

Comment:
nit: like above, remove `if`: rename more declarative, e.g. `signed in student with course progress is redirected to next lesson`